### PR TITLE
Add Abs, Dot and AbsDot and respective tests to Vector4

### LINKED ISSUE #71: Add functions from Vector3 to Vector2 and Vector4
**Original report ([archived issue](https://osrf-migration.github.io/ignition-gh-pages/#!/ignitionrobotics/ign-math/issues/71)) by Louise Poubel (Bitbucket: [chapulina](https://bitbucket.org/%7B5cfa2075-477b-4ded-bdb9-8d2479544ec4%7D/), GitHub: [chapulina](https://github.com/chapulina)).**

----------------------------------------
For example, there is a [Vector3::Normalized](https://github.com/ignitionrobotics/ign-math/blob/05458c89f0434e2774f19995e713ed7dd94f1018/include/ignition/math/Vector3.hh#L143), but no such function for [Vector2](https://github.com/ignitionrobotics/ign-math/blob/05458c89f0434e2774f19995e713ed7dd94f1018/include/ignition/math/Vector2.hh) or [Vector4](https://github.com/ignitionrobotics/ign-math/blob/05458c89f0434e2774f19995e713ed7dd94f1018/include/ignition/math/Vector4.hh). The same goes for `Round`, `Rounded` and other functions.

IssueForNewDevelopers

### PR DESCRIPTION
**Pull request by luccosta**
Resolves #71 .

This implements the functions Abs(), Dot(Vector4) and AbsDot(Vector4) in Vector4 API, similar to the existing same functions in Vector3.

It enables Vector4 objects to perform:

```
vec4.Abs();
vec4_a.Dot(vec4_b);
vec4_a.AbsDot(vec4_b);
```

Did all checks and tests.

### CODE REVIEW

File: src/Vector4_TEST.cc [hunk 1]
```diff
@@ -61,6 +61,16 @@ TEST(Vector4dTest, Vector4d)
...
 
+  // ::DotProd
+  EXPECT_TRUE(math::equal(60.0, v.Dot(math::Vector4d(1, 2, 3, 4)), 1e-2));
+
+  // ::AbsDotProd
+  v1.Set(-1, -2, -3, -4);
+  EXPECT_TRUE(math::equal(60.0, v.AbsDot(v1), 1e-2));
+
+  // ::GetAbs
+  EXPECT_TRUE(v1.Abs() == math::Vector4d(1, 2, 3, 4));
```
Comment (mjcarroll) [inline src/Vector4_TEST.cc:72]: Maybe a test of `Abs` with a vector with positive values (for completeness).

Comment (luccosta) [reply src/Vector4_TEST.cc:72]: Great! Added in the last commit.

**Comment by luccosta** [comment]
Locally everything is running ok, but the Ubuntu Bionic CI is getting:

```
Unexpected input(s) 'script-after-make', valid inputs are ['entryPoint', 'args', 'github_token', 'apt-dependencies', 'codecov-token', 'cmake-args']
```

And I'm not finding anything about this.

**Comment by chapulina** [comment]
> Unexpected input(s) 'script-after-make'

Don't worry, you can ignore this. It's a problem with our infrastructure.

**Pull request merged**